iotilecore/iotile/core/hw/auth/inmemory_auth_provider.py

```diff
@@ -0,0 +1,63 @@
+from iotile.core.exceptions import NotFoundError
+from .rootkey_auth_provider import RootKeyAuthProvider
+from .auth_provider import AuthProvider
+
+
+class InMemoryAuthProvider(RootKeyAuthProvider):
+    """Key provider implementation that allows user to set a password in memory"""
+
+    _shared_passwords = {}
```

Comment:
@timburke Wondering if we should limit the size of `_shared_passwords`...

Reply:
No, it can't fill up on its own so we shouldn't limit the user in how many entries they choose to put there.